bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitLightbulbImpl.java

```diff
@@ -12,20 +12,58 @@
  */
 package org.openhab.io.homekit.internal.accessories;
 
-import org.eclipse.smarthome.core.items.ItemRegistry;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import org.eclipse.smarthome.core.items.GenericItem;
+import org.eclipse.smarthome.core.items.GroupItem;
 import org.eclipse.smarthome.core.library.items.SwitchItem;
+import org.eclipse.smarthome.core.library.types.OnOffType;
 import org.openhab.io.homekit.internal.HomekitAccessoryUpdater;
+import org.openhab.io.homekit.internal.HomekitCharacteristicType;
+import org.openhab.io.homekit.internal.HomekitSettings;
 import org.openhab.io.homekit.internal.HomekitTaggedItem;
+import io.github.hapjava.accessories.LightbulbAccessory;
+import io.github.hapjava.characteristics.HomekitCharacteristicChangeCallback;
+import io.github.hapjava.services.impl.LightbulbService;
 
 /**
  * Implements Lightbulb using an Item that provides an On/Off state
  *
  * @author Andy Lintner - Initial contribution
  */
-class HomekitLightbulbImpl extends AbstractHomekitLightbulbImpl<SwitchItem> {
+class HomekitLightbulbImpl extends AbstractHomekitAccessoryImpl implements LightbulbAccessory  {
 
-    public HomekitLightbulbImpl(HomekitTaggedItem taggedItem, ItemRegistry itemRegistry,
-            HomekitAccessoryUpdater updater) {
-        super(taggedItem, itemRegistry, updater, SwitchItem.class);
+    public HomekitLightbulbImpl(HomekitTaggedItem taggedItem, List<HomekitTaggedItem> mandatoryCharacteristics,
+            HomekitAccessoryUpdater updater, HomekitSettings settings) throws IncompleteAccessoryException {
+        super(taggedItem, mandatoryCharacteristics, updater, settings);
+        this.getServices().add(new LightbulbService(this));
     }
+
+    @Override
+    public CompletableFuture<Boolean> getLightbulbPowerState() {
+        OnOffType state = getStateAs(HomekitCharacteristicType.ON_STATE, OnOffType.class);
+        return CompletableFuture.completedFuture(state != null && state == OnOffType.ON);
```

Comment:
```suggestion
        return CompletableFuture.completedFuture(state != null && state == OnOffType.ON);
```

If it's ON, it cannot be null.

Reply:
right.